zerver/views/registration.py

```diff
@@ -52,9 +52,12 @@
 def check_prereg_key_and_redirect(request: HttpRequest, confirmation_key: str) -> HttpResponse:
     # If the key isn't valid, show the error message on the original URL
     confirmation = Confirmation.objects.filter(confirmation_key=confirmation_key).first()
-    prereg_user = confirmation.content_object
+    # if the prereg user has status revoked the confirmation link should throw an error page
+    confirmation_status_obj = Confirmation.objects.get(confirmation_key=confirmation_key,
+                                                       type=confirmation.type)
+    prereg_user = confirmation_status_obj.content_object
     revoked_value = confirmation_settings.STATUS_REVOKED
-    if prereg_user.status == revoked_value:
+    if hasattr(prereg_user, "status") and prereg_user.status == revoked_value:
```

Comment:
There's not need to check is prereg_user has ``status``, it always has it, that's how the model is defined.